rfc-079-script-imports-of-production-data.md

```diff
@@ -0,0 +1,67 @@
+# Script imports of production data
+
+## Summary
+
+Developers writing production data to file on a local or remote server MUST do
+this via a script. It MUST NOT be done by manual actions, such as via the Rails
+console or `pg_dump`.
+
+This script MUST anonymise any sensitive data, and the data dump MUST be stored
+outside of any Git repositories to ensure it can't be inadvertently committed
+alongside any changes to the codebase.
+
+## Problem
+
+Fetching data from a database by hand is error-prone. A data dump may be left
+on a server, committed to a repository or even in some cases alter production
+data in unforeseen ways. Developers may not realise they have exposed or
+impacted production data until long after an incident occurs.
+
+By writing a script that developers can run from their local machine to
+automate this process, we ensure that every import of data from a production
+environment is done in the same way, minimising room for error. It also allows
+us to scrub the resulting data dump of any personally-identifying and
+security-sensitive information.
+
+Committing a data-fetching script to the repository makes it visible to other
+members of the team in the code review process, allows it to be developed
+against a local instance of the database, and allows others to provide feedback
+to ensure we're reaching a consensus on the safest way of fetching this data.
+
+## Proposal
+
+If developing an application requires a developer to write any production data
+containing sensitive data to a local file or a file on a remove server, this
+MUST be done via a script.
+
+The script MUST anonymise any security-sensitive data. It SHOULD remove or
+anonymise any personal data where relevant. Anonymisation SHOULD be done
+before any data is written to disc. This MAY be done on the remote server, or
+piped through a script that substitutes any sensitive values before writing to
+a file.
```

Comment:
Should we upgrade those SHOULDs to MUSTs? We can recommend replacing personal information using eg Faker, if you still need "people" in your dump.

Reply:
I think we'd need to make it "MUST … unless", rather than a straight MUST. For example, a script that dumps out records from eLinks to share with the eJudiciary team will be full of personal data. Anonymised it's useless.

Reply:
Might that also need some of the sensitive data, too?

Reply:
Given the definition of SHOULD, I'm happy with it for this point:

> This word, or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.

That said, I think I'd drop the "where relevant" from the clause.